### PR TITLE
Modify url in git_clone

### PR DESCRIPTION
**Pull request by lukaszachy**
For now just accomodate git:// deprecation in RH.

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -2861,8 +2861,18 @@ def public_git_url(url: str) -> str:
     :returns: URL that is publicly accessible without authentication,
         or the original URL if no applicable conversion was found.
     """
+    return modify_git_url(url, PUBLIC_GIT_URL_PATTERNS)
 
-    for pattern, replacement in PUBLIC_GIT_URL_PATTERNS:
+
+def modify_git_url(url: str, patterns: List[Tuple[str, str]]) -> str:
```
Comment (happz) [inline tmt/utils.py:2867]: Maybe `rewrite_git_url`? URL rewriting is a common topic, and it's what the function does.

File: tmt/utils.py [hunk 2]
```diff
@@ -2861,8 +2861,18 @@ def public_git_url(url: str) -> str:
...
+    """
+    Modify git url based on supplied patterns
+
+    :param url: an URL to modify
+    :param patterns: List of pattern to try in order
```
Comment (happz) [inline tmt/utils.py:2872]: "patternS"

File: tmt/utils.py (hunk 1 above)
Comment (lukaszachy) [reply tmt/utils.py]: fixed in [de05583](https://github.com/teemtee/tmt/pull/2223/commits/de055837d6454899dacf47e2a499926e630ab043)

File: tmt/utils.py (hunk 2 above)
Comment (lukaszachy) [reply tmt/utils.py]: fixed in [de05583](https://github.com/teemtee/tmt/pull/2223/commits/de055837d6454899dacf47e2a499926e630ab043)

**Review by happz — APPROVED** [review]

**Pull request merged**